solr/modules/opentelemetry/src/test/org/apache/solr/opentelemetry/OtelTracerConfiguratorTest.java

```diff
@@ -73,4 +86,24 @@ public void testSetDefaultIfNotConfigured() {
     assertEquals("default", instance.getCurrentOtelConfig().get("OTEL_YEY"));
     assertEquals("prop-k1", instance.getCurrentOtelConfig().get("OTEL_K1"));
   }
+
+  @Test
+  public void testInjected() throws Exception {
+    // Make sure the batch exporter times out before our thread lingering time of 10s
+    instance.setDefaultIfNotConfigured("OTEL_BSP_SCHEDULE_DELAY", "1000");
+    instance.setDefaultIfNotConfigured("OTEL_BSP_EXPORT_TIMEOUT", "2000");
+    MiniSolrCloudCluster cluster =
+        new MiniSolrCloudCluster.Builder(2, createTempDir())
+            .addConfig("config", TEST_PATH().resolve("collection1").resolve("conf"))
+            .withSolrXml(getFile("solr/solr.xml").toPath())
+            .build();
+    try {
+      TimeOut timeOut = new TimeOut(2, TimeUnit.MINUTES, TimeSource.NANO_TIME);
+      timeOut.waitFor(
+          "Waiting for GlobalTracer is registered",
+          () -> GlobalTracer.get().toString().contains("TracerShim"));
```

Comment:
this is surprising... is the shim installed in some async thread?

Reply:
What's surprising? That we need to wait for the global tracer to be registered? Guess we're waiting for the `MiniSolrCloudCluster` to bootstrap, which will eventually lead to CoreContainer registering the tracer. This test was copied from Jaeger module.

Reply:
Yeah.  MiniSolrCloudCluster is certainly finished when build is returned.  Look at the end of MSCC's constructor -- it waits for all servers to be available.

Reply:
When running this test locally, it certainly executes fast. Perhaps MiniSolrCloudCluster behaved differently back in 2019 when Jaeger was added? Anyway, I removed the timeout stuff.